### PR TITLE
Simplify methods and functions in `live-migration-source` file

### PR DESCRIPTION
**Pull request by VirrageS**
**What this PR does / why we need it**: This MR simplifies some expressions which are too verbose and sometimes even confusing with simpler ones for readability. Also removes unused parameters.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*: None

**Special notes for your reviewer**: None

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-launcher/virtwrap/live-migration-source.go [hunk 1]
```diff
@@ -409,13 +408,7 @@ func (l *LibvirtDomainManager) startMigration(vmi *v1.VirtualMachineInstance, op
...
 	}
 
-	err = l.asyncMigrate(vmi, options)
-	if err != nil {
-		log.Log.Object(vmi).Reason(err).Error(liveMigrationFailed)
-		l.setMigrationResult(vmi, true, fmt.Sprintf("%v", err), "")
-		return err
-	}
-
+	l.asyncMigrate(vmi, options)
```
Comment (xpivarc) [inline pkg/virt-launcher/virtwrap/live-migration-source.go:411]: IMHO we could do even better. `go l.migrate(vmi, options)` would be more idiomatic

**Review by xpivarc — COMMENTED** [review]
/approve
/cc @vladikr @jean-edouard

File: pkg/virt-launcher/virtwrap/live-migration-source.go (hunk 1 above)
Comment (VirrageS) [reply pkg/virt-launcher/virtwrap/live-migration-source.go:411]: Makes sense, thanks! Fixed.

File: pkg/virt-launcher/virtwrap/live-migration-source.go [hunk 2]
```diff
@@ -1011,8 +987,7 @@ func shouldImmediatelyFailMigration(vmi *v1.VirtualMachineInstance) bool {
 	return shouldFail
 }
 
-func (l *LibvirtDomainManager) asyncMigrate(vmi *v1.VirtualMachineInstance, options *cmdclient.MigrationOptions) error {
-
+func (l *LibvirtDomainManager) migrate(vmi *v1.VirtualMachineInstance, options *cmdclient.MigrationOptions) {
```
Comment (xpivarc) [inline pkg/virt-launcher/virtwrap/live-migration-source.go:990]: I think getting rid of `go func()` and moving it(the `go` keyword) to caller could be more clear & idiomatic.

**Review by xpivarc — COMMENTED** [review]
/approve
Thank you for the clean-up!

**Comment by xpivarc** [comment]
/hold 
Test failures are unrelated. An investigation is ongoing.

**Review by jean-edouard — APPROVED** [review]

**Pull request merged**